lib/mkFloxEnv.nix

```
@@ -6,14 +6,14 @@
 }: {
   system,
   modules,
+  context,
 }:
 (lib.evalModules {
   modules =
     [
       {
         _module.args = {
-          pkgs = nixpkgs.legacyPackages.${system};
-          inherit system self;
+          inherit system context;
```

```suggestion
         context = context // (context.context' system);
```

Since you want a systemized context anyway, you can just construct it here and pass that into the module.